File: android/SmartIRHub/app/src/main/java/com/ms8/smartirhub/android/_tests/dev_playground/remote_layout/asymmetricgridview/AGVBaseAdapter.java
package com.ms8.smartirhub.android._tests.dev_playground.remote_layout.asymmetricgridview;

import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;

interface AGVBaseAdapter<T extends RecyclerView.ViewHolder> {
  int getActualItemCount();
  AsymmetricItem getItem(int position);
  void notifyDataSetChanged();
  int getItemViewType(int actualIndex);
  AsymmetricViewHolder<T> onCreateAsymmetricViewHolder(int position, ViewGroup parent, int viewType);
  void onBindAsymmetricViewHolder(AsymmetricViewHolder<T> holder, ViewGroup parent, int position);
}

File: android/SmartIRHub/app/src/main/java/com/ms8/smartirhub/android/_tests/dev_playground/remote_layout/asymmetricgridview_k/AGVBaseAdapter.kt
package com.ms8.smartirhub.android._tests.dev_playground.remote_layout.asymmetricgridview_k

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

interface AGVBaseAdapter<T : RecyclerView.ViewHolder> {
    val actualItemCount: Int
    fun getItem(position: Int): AsymmetricItem
    fun notifyDataSetChanged()
    fun getItemViewType(actualIndex: Int): Int
    fun onCreateAsymmetricViewHolder(position: Int, parent: ViewGroup, viewType: Int): AsymmetricViewHolder<T>
    fun onBindAsymmetricViewHolder(holder: AsymmetricViewHolder<*>, parent: ViewGroup, position: Int)
}